src/adapter/graphql/tests/tests/test_gql_dataset_flow_configs.rs

```diff
@@ -0,0 +1,730 @@
+                            setConfigCronExpression (
+                                datasetFlowType: "<dataset_flow_type>",
+                                paused: <paused>,
+                                cronExpression: "<cron_expression>"
+                            ) {
+                                __typename,
+                                paused
+                                schedule {
+                                    __typename
+                                    ... on CronExpression {
+                                        cronExpression
+                                    }
+                                }
+                                batching {
+                                    __typename
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            "#
+        )
+        .replace("<id>", &id.to_string())
+        .replace("<dataset_flow_type>", dataset_flow_type)
+        .replace("<paused>", if paused { "true" } else { "false" })
+        .replace("<cron_expression>", cron_expression)
+    }
+
+    fn set_config_batching_mutation(
+        id: &DatasetID,
+        dataset_flow_type: &str,
+        paused: bool,
+        throttling_period: (u64, &str),
+        minimial_data_batch: u64,
+    ) -> String {
+        indoc!(
+            r#"
+            mutation {
+                datasets {
+                    byId (datasetId: "<id>") {
+                        flowConfigs {
+                            setConfigBatching (
+                                datasetFlowType: "<dataset_flow_type>",
+                                paused: <paused>,
+                                throttlingPeriod: { every: <every>, unit: "<unit>" },
+                                minimalDataBatch: <min_data_batch>
+                            ) {
+                                __typename,
+                                paused
+                                schedule {
+                                    __typename
+                                }
+                                batching {
+                                    __typename
+                                    throttlingPeriod {
+                                        every
+                                        unit
+                                    }
+                                    minimalDataBatch
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            "#
+        )
+        .replace("<id>", &id.to_string())
+        .replace("<dataset_flow_type>", dataset_flow_type)
+        .replace("<paused>", if paused { "true" } else { "false" })
+        .replace("<every>", &throttling_period.0.to_string())
+        .replace("<unit>", throttling_period.1)
+        .replace("<min_data_batch>", &minimial_data_batch.to_string())
```

Comment:
reminds me of `jinja` :)

Reply:
It basically is, but as you may have noticed, it greately reduced the size of the tests. Each of these templates are repeating at least 3 times.